app/src/main/java/io/gnosis/safe/ui/signing/owners/OwnerListAdapter.kt

```diff
@@ -0,0 +1,116 @@
+package io.gnosis.safe.ui.signing.owners
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.paging.PagingDataAdapter
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import io.gnosis.safe.databinding.ItemOwnerSelectionHeaderBinding
+import io.gnosis.safe.databinding.ItemOwnerSelectionOwnerBinding
+import io.gnosis.safe.ui.base.adapter.UnsupportedViewType
+import io.gnosis.safe.utils.formatEthAddress
+import pm.gnosis.model.Solidity
+import pm.gnosis.svalinn.common.utils.visible
+import java.lang.ref.WeakReference
+
+class OwnerListAdapter() : PagingDataAdapter<Solidity.Address, RecyclerView.ViewHolder>(COMPARATOR) {
+
+    var pagesVisible = 1
+    private var selectedOwnerPosition: Int = 1
+
+    private var listener: WeakReference<OnOwnerItemClickedListener>? = null
+
+    fun setListener(listener: OnOwnerItemClickedListener) {
+        this.listener = WeakReference(listener)
```

Comment:
Refreshing my memory about WeakReferences...

Reply:
I think I got it. When the Fragment is garbage collected. So will the listener reference. 👍